samples/SocialSample/Startup.cs

```diff
@@ -199,13 +206,10 @@ 127.0.0.1 MsSecSample.localhost.this
                         await context.Response.WriteAsync("<a href=\"?authscheme=" + type.AuthenticationScheme + "\">" + (type.Caption ?? "(suppressed)") + "</a><br>");
                     }
                     await context.Response.WriteAsync("</body></html>");
-                });
-            });
-
-            // Sign-out to remove the user cookie.
-            app.Map("/logout", signoutApp =>
-            {
-                signoutApp.Run(async context =>
+                    return;
+                }
+                
+                if (context.Request.Path.StartsWithSegments(new PathString("/logout")))
                 {
                     await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
                     context.Response.ContentType = "text/html";
```

Comment:
I would argue calling these methods which immediately do something is easier to grok, than the old behavior of stuff happens at some undetermined future time (whenever the response is sent).  Perhaps you are just used to the old behavior, and the new will grow on you.  I didn't play with the old behavior much (so I'm missing any fond memories of it)